src/utils/usePledge.ts

```diff
@@ -0,0 +1,117 @@
+import { ipcRenderer } from "electron";
+import { GraphQLClient } from "graphql-request";
+import { getSdk } from "src/generated/graphql-request";
+import { NodeInfo } from "src/config";
+import { ActivationFunction, ActivationStep } from "src/interfaces/activation";
+import { useTx } from "src/utils/useTx";
+import { useStore } from "./useStore";
+import { useTransactionResultLazyQuery } from "src/generated/graphql";
+import { sleep } from "src/utils";
+
+export function usePledge() {
+  const tx = useTx();
+  const account = useStore("account");
+
+  const [fetchStatus, { loading, data: txState, stopPolling }] =
+    useTransactionResultLazyQuery({
+      pollInterval: 1000,
+      fetchPolicy: "no-cache",
+    });
+
+  const activate: ActivationFunction = async () => {
+    let step: ActivationStep = "getGraphQLClient";
+    const nodeInfo: NodeInfo = await ipcRenderer.invoke("get-node-info");
+    const sdks = getSdk(
+      new GraphQLClient(
+        `http://${nodeInfo.host}:${nodeInfo.graphqlPort}/graphql`
+      )
+    );
+
+    try {
+      step = "preflightCheck";
+      if (!account.loginSession) {
+        throw new Error("No private key");
+      }
+      const { data } = await sdks.CheckContracted({
+        agentAddress: account.loginSession.address.toHex(),
+      });
+      const { approved, patronAddress } = data.stateQuery.pledge;
+
+      if (!approved) {
+        if (patronAddress === null) {
+          step = "checkRequestPledge";
+          for (let i = 0; i <= 60; i++) {
+            const { data } = await sdks.CheckContracted({
+              agentAddress: account.loginSession.address.toHex(),
+            });
+            if (data.stateQuery.pledge.patronAddress !== null) {
+              break;
+            }
+            if (i === 60) throw new Error("Contract Check Timeout.");
+            sleep(1000); // timeout 1 minutes
```

Comment:
We might need to refactor this timeout handling in the future. 
(it's hard to guess the entire structure because constants(i.e., `1000`, `60`) were spread on `for`, `check`, `sleep`)

Reply:
I rewrote them with Promise with setTimeout and setInterval.

Reply:
It works with blocking. resolved.